### PR TITLE
[docs] Simulator python bindings

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

This PR adds some doc-strings to Simulator python bindings and polishes some related docs.

## How Has This Been Tested

Local doc build.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/esp/bindings/SimBindings.cpp [hunk 1]
```diff
@@ -84,97 +84,169 @@ void initSimBindings(py::module& m) {
...
-           "scene_id"_a = 0)
+           "scene_id"_a = 0, R"(
+        Remove an object instance from the scene. Optionally leave its root SceneNode and visual SceneNode on the SceneGraph.
+
+        .. note-warning::
+
+            Removing an object which was attached to the SceneNode
+            of an Agent expected to continue producing observations
+            will likely result in errors if delete_object_node is true.
+            )")
```
Comment (aclegg3) [inline src/esp/bindings/SimBindings.cpp:131]: @mosra Not sure why this isn't generating a method detail block with a warning note. I tried copying the white-space formatting for the `semantic_scene` (ln 62). Any thoughts?

Comment (mosra) [reply src/esp/bindings/SimBindings.cpp:131]: Any warning printed (if you can see it among all the others)? What does it generate instead?

Comment (Skylion007) [reply src/esp/bindings/SimBindings.cpp:131]: You can try tools like rstcheck to further validate it / find the specific error.

Comment (aclegg3) [reply src/esp/bindings/SimBindings.cpp:131]: It generates the brief doc, but no entry in the method Documentation section. I can't find any warnings about this either.
image

Comment (erikwijmans) [reply src/esp/bindings/SimBindings.cpp:131]: You may need to un-indent things.  I am not sure how the rst parser feels about all that whitespace.

**Review by Skylion007 — APPROVED** [review]

**Review by Skylion007 — APPROVED** [review]

**Pull request merged**